src/panel/pages/timeline/components/TimelineTooltip.tsx

```diff
@@ -42,59 +55,103 @@ const TooltipElement = styled.p<{ position?: TooltipPosition }>`
     border-left: 6px solid transparent;
     border-right: 6px solid transparent;
     margin-top: -1px;
-    left: 50%;
+    left: calc(50% - ${(p) => p.offset}px);
     top: 100%;
     transform: translate(-50%, 0);
   }
 `;
 
 export const useTooltip = () => {
-  const ref: MutableRefObject<HTMLElement | null> = useRef<HTMLElement>(null);
+  const containerRef: MutableRefObject<HTMLElement | null> = useRef<
+    HTMLElement
+  >(null);
+  const tooltipRef: MutableRefObject<HTMLElement | null> = useRef<HTMLElement>(
+    null
+  );
+  const tooltipOffset: MutableRefObject<number> = useRef(0);
+  const mouseX = useRef<number | undefined>(undefined);
   const [hasRef, setHasRef] = useState(false);
   const [isVisible, setIsVisible] = useState(false);
   const [tooltipProps, setTooltipProps] = useState<
-    ComponentProps<typeof TimelineTooltip>
+    ComponentProps<typeof TimelineTooltip> & TimelineTooltipProps
   >({});
 
   const calculateTooltipPosition = useCallback(() => {
-    if (!ref.current) {
+    if (!containerRef.current) {
       return;
     }
 
-    const { x, y, width } = ref.current.getBoundingClientRect(); //eslint-disable-line @typescript-eslint/no-non-null-assertion
-    setTooltipProps({
+    const { x, y, width } = containerRef.current.getBoundingClientRect(); //eslint-disable-line @typescript-eslint/no-non-null-assertion
+
+    const newTooltipProps: ComponentProps<typeof TimelineTooltip> &
+      TimelineTooltipProps = {
       style: {
         position: "fixed",
         left: x + width / 2,
         top: y,
         transform: `translateX(-50%) translateY(-100%)`,
+        paddingBottom: 8,
+        zIndex: 9,
```

Comment:
I don't think the zIndex is needed as we don't use zIndexing anywhere else on the site and `position: fixed` should overlay.

Reply:
Without it we seem to get some slight funky behaviour with the TimelineRow/TimelineEvent overlaying the Tooltips and the TimelineIcons. There may be another alternative though?

![image](https://user-images.githubusercontent.com/35961363/78122997-6cf6e600-7405-11ea-9859-9ad4b16bced4.png)